# PhD thesis title: EPIGENETIC DYSREGULATION IN LYMPHOID LEUKEMIAS
Over the last two decades, large scale epigenomic studies showed that aberrant epigenomic landscape is an essential molecular feature of cancer cells and established as a hallmark of cancer. Epigenetic dysregulation in cancer could arise due to genetic and epigenetic changes of the proteins involved in epigenetic modifications. The goal of my study is to characterize dysregulated epigenomic states and understand their mechanisms in two different lymphoid leukemias e.g., T-cell acute lymphoblastic leukemia (T-ALL) and B-cell chronic lymphocytic leukemia (CLL). Using integrative analysis of multiple layers of epigenomic datasets, my analysis identified disease specific enhancers which are associated with proto-oncogene activation in those leukemias. I identified pathogenic enhancers are established and maintained by the runt related transcription factor 1 (RUNX1) in leukemia. To leverage the valuable epigenetic datasets in clinical subtypes of CLL, I developed a computational pipeline, CRIS, to classify CLL patients based on somatic hypermutations of immunoglobulin variable gene using RNA-seq. 

#### Source code

# Commands and parameters for tools

# Chapter 1: data analysis and visualization

# Chapter 2: data analysis and visualization

# Chapter 3: data analysis and visualization

#### Data
# Chapter 1: 
- Raw sequencing data is available 
- Processed files: bed files and RPKM matrices of gene expression. 

# Chapter 2: 
- Raw sequencing data is available at CEMT and Blueprint.
- Pipeline: https://github.com/Rashedul/CRIS 

# Chapter 3: 
- Raw sequencing data is available at CEMT and Blueprint.
- Processed files: bed files and RPKM matrices of gene expression. 

#### Publications 

