## Thesis title: EPIGENETIC DYSREGULATION IN LYMPHOID LEUKEMIAS
Over the last two decades, large scale epigenomic studies showed that aberrant epigenomic landscape is an essential molecular feature of cancer cells and established as a hallmark of cancer. Epigenetic dysregulation in cancer could arise due to genetic and epigenetic changes of the proteins involved in epigenetic modifications. The goal of my study is to characterize dysregulated epigenomic states and understand their mechanisms in two different lymphoid leukemias e.g., T-cell acute lymphoblastic leukemia (T-ALL) and B-cell chronic lymphocytic leukemia (CLL). Using integrative analysis of multiple layers of epigenomic datasets, my analysis identified disease specific enhancers which are associated with proto-oncogene activation in those leukemias. I identified pathogenic enhancers are established and maintained by the runt related transcription factor 1 (RUNX1) in leukemia. To leverage the valuable epigenetic datasets in clinical subtypes of CLL, I developed a computational pipeline, CRIS, to classify CLL patients based on somatic hypermutations of immunoglobulin variable gene using RNA-seq. 

## Source code

#### [Commands and parameters for bioinformatic tools](https://github.com/Rashedul/PhD_thesis/blob/main/Command.md)

#### Chapter 1: Introduction

#### Chapter 2: RUNX1 colludes with NOTCH1 to reprogram chromatin in T-cell lymphoblastic leukemia
- [data analysis and visualization](https://github.com/Rashedul/PhD_thesis/blob/main/script/chapter-3.md)

#### Chapter 3: CRIS: Complete Reconstruction of Immunoglobulin V-D-J Sequences from RNA-seq data
- [data analysis and visualization](https://github.com/Rashedul/PhD_thesis/blob/main/script/chapter-3.md)

#### Chapter 4: Aberrant chromatin landscape in chronic lymphocytic leukemia
- [data analysis and visualization](https://github.com/Rashedul/PhD_thesis/blob/main/script/chapter-3.md)

#### Chapter 5: Conclusion 

## Data
#### Chapter 2: 
- Raw sequencing data is available via accession number GSE99706.
- Processed files: [bed files, super-enhanceers, differentially expressed genes and RPKM matrix of gene expression.](https://github.com/Rashedul/PhD_thesis/tree/main/data/chapter-2) 

#### Chapter 3: 
- Raw sequencing data is available at [CEMT](https://thisisepigenetics.ca/data/CEMT/epi2021/grid/) and [BLUEPRINT](http://dcc.blueprint-epigenome.eu/#/home).
- Pipeline: https://github.com/Rashedul/CRIS 

#### Chapter 4: 
- Raw sequencing data is available at [CEMT](https://thisisepigenetics.ca/data/CEMT/epi2021/grid/) and [BLUEPRINT](http://dcc.blueprint-epigenome.eu/#/home).
- Processed files: [bed files, super-enhanceers, differentially expressed genes and RPKM matrix of gene expression.](https://github.com/Rashedul/PhD_thesis/tree/main/data/chapter-4)

## [Thesis at UBC library](https://open.library.ubc.ca/) 

## Publications 

- Catherine E. Jenkins, Samuel Gusscott, Rachel J. Wong, Olena O. Shevchuk, Gurneet Rana, Vincenzo Giambra, Kateryna Tyshchenko, **Rashedul Islam**, Martin Hirst, Andrew P. Weng. [RUNX1 promotes cell growth in human T-cell acute lymphoblastic leukemia by transcriptional regulation of key target genes](https://pubmed.ncbi.nlm.nih.gov/29733873/). *Experimental Hematology*. 2018 Aug; 64:84-96. 

